### PR TITLE
fix(TDP-2734): use label instead of name

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2734

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

File: dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header.html [hunk 1]
```diff
@@ -19,8 +19,8 @@
             <div class="grid-header-title" title="{{ lookupDatagridHeaderCtrl.column.name }}">{{ lookupDatagridHeaderCtrl.column.name }}</div>
         </div>
 
-        <div class="grid-header-type" ng-if="lookupDatagridHeaderCtrl.column.domain">{{ lookupDatagridHeaderCtrl.column.domain }}</div>
-        <div class="grid-header-type" ng-if="(!lookupDatagridHeaderCtrl.column.domain)">{{ lookupDatagridHeaderCtrl.converterService.simplifyType(lookupDatagridHeaderCtrl.column.type) }}</div>
+        <div class="grid-header-type" ng-if="lookupDatagridHeaderCtrl.column.domainLabel">{{ lookupDatagridHeaderCtrl.column.domainLabel }}</div>
+        <div class="grid-header-type" ng-if="(!lookupDatagridHeaderCtrl.column.domainLabel)">{{ lookupDatagridHeaderCtrl.converterService.simplifyType(lookupDatagridHeaderCtrl.column.type) }}</div>
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header.html]: {{ lookupDatagridHeaderCtrl.column.domainLabel ? lookupDatagridHeaderCtrl.column.domainLabel :  lookupDatagridHeaderCtrl.converterService.simplifyType(lookupDatagridHeaderCtrl.column.type) }}

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js [hunk 2]
```diff
@@ -96,8 +96,14 @@ export default class TypeTransformMenuCtrl {
...
 	 */
 	_refreshCurrentDomain() {
-		this.currentDomain = this.column.domain ? this.column.domain : this.column.type.toUpperCase();
-		this.currentSimplifiedDomain = this.column.domain ? this.column.domain : this.ConverterService.simplifyType(this.column.type);
+		if (this.column.domain && this.column.domainLabel) {
+			this.currentDomain = this.column.domainLabel;
+			this.currentSimplifiedDomain = this.column.domainLabel;
+		}
+		else {
+			this.currentDomain = this.column.type.toUpperCase();
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js]: Please use CSS for that if needed

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js [hunk 3]
```diff
@@ -96,8 +96,14 @@ export default class TypeTransformMenuCtrl {
 	 * @description Refreshes current domain and simplified domain variables
 	 */
 	_refreshCurrentDomain() {
-		this.currentDomain = this.column.domain ? this.column.domain : this.column.type.toUpperCase();
-		this.currentSimplifiedDomain = this.column.domain ? this.column.domain : this.ConverterService.simplifyType(this.column.type);
+		if (this.column.domain && this.column.domainLabel) {
+			this.currentDomain = this.column.domainLabel;
+			this.currentSimplifiedDomain = this.column.domainLabel;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js]: Same value, please use constant, DRY

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/418/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/418/)

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js (hunk 2 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js]: Done, thanks

File: dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header.html (hunk 1 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header.html]: I added a getter in both `lookup-datagrid-header.html` and `datagrid-header.html` to solve that.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/420/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/420/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/427/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/427/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/431/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/431/)

File: dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header-controller.js [hunk 4]
```diff
@@ -50,4 +50,17 @@ export default function LookupDatagridHeaderCtrl(ConverterService, state, StateS
...
 	};
+
+	/**
+     * @ngdoc method
+     * @name getTypeLabel
+     * @methodOf data-prep.lookup-datagrid-header.controller:LookupDatagridHeaderCtrl
+     * @description returns the type label
+     */
+	vm.getTypeLabel = function () {
+		if (vm.column.domainLabel) {
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header-controller.js]:
```javascript
return vm.column.domainLabel || vm.converterService.simplifyType(vm.column.type);
```

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-component.spec.js [hunk 5]
```diff
@@ -125,11 +135,20 @@ describe('Transformation menu component', () => {
...
 		expect(items.eq(2).hasClass('divider')).toBe(true);
 
-		expect(items.eq(3).text().trim()).toBe('Set as STRING');
-		expect(items.eq(4).text().trim()).toBe('Set as INTEGER');
-		expect(items.eq(5).text().trim()).toBe('Set as DECIMAL');
-		expect(items.eq(6).text().trim()).toBe('Set as BOOLEAN');
-		expect(items.eq(7).text().trim()).toBe('Set as DATE');
+		for (let i = 3; i < 7; i++) {
+			expect(items.eq(i).find('span').eq(0).text().trim()).toBe(TRANSLATIONS.COLUMN_TYPE_SET);
+			expect(items.eq(i).find('span').eq(1).text().trim()).toBe(types[i - 3]);
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-component.spec.js]: DRY, dot not parse DOM N times
```javascript
const offset = 3; 
types.forEach((type, index) => {
	const itemElementSpans = items.eq(index + offset).find('span');
	expect(itemElementSpans.eq(0).text().trim()).toBe(TRANSLATIONS.COLUMN_TYPE_SET);
	expect(itemElementSpans.eq(1).text().trim()).toBe(type);
}
```

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js [hunk 6]
```diff
@@ -96,8 +96,15 @@ export default class TypeTransformMenuCtrl {
...
 	_refreshCurrentDomain() {
-		this.currentDomain = this.column.domain ? this.column.domain : this.column.type.toUpperCase();
-		this.currentSimplifiedDomain = this.column.domain ? this.column.domain : this.ConverterService.simplifyType(this.column.type);
+		if (this.column.domain && this.column.domainLabel) {
+			const label = this.column.domainLabel;
+			this.currentDomain = label;
+			this.currentSimplifiedDomain = label;
+		}
+		else {
+			this.currentDomain = this.column.type.toUpperCase();
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js]: Please, do not use hard coded case instructions if it's just for display purpose

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html [hunk 7]
```diff
@@ -19,15 +19,22 @@
...
             ng-click="$ctrl.changeType(type)">
-            <input type="radio" value="{{ ::type.id}}" ng-model="$ctrl.currentDomain"
-                   ng-checked="$ctrl.shouldBeChecked(type)"/>
-            <span translate-once="COLUMN_TYPE_SET"></span><span>&nbsp;</span>{{::type.labelKey | translate }}
+            <input  type="radio"
+                    value="{{ ::type.id}}"
+                    ng-model="$ctrl.currentDomain"
+                    ng-checked="$ctrl.shouldBeChecked(type)"
+            />
+            <span class="set-type-label" translate-once="COLUMN_TYPE_SET"></span><span class="type">{{::type.labelKey | translate }}</span>
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: you could use translate-once directive

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html [hunk 8]
```diff
@@ -19,15 +19,22 @@
     <ul class="submenu">
         <li ng-repeat="semanticDomain in $ctrl.domains track by semanticDomain.id"
             ng-click="$ctrl.changeDomain(semanticDomain)">
-            <input type="radio" value="{{ ::semanticDomain.id}}" ng-model="$ctrl.currentDomain"/>
-            {{ ::semanticDomain.label | translate }} {{ semanticDomain.frequency }} %
+            <input  type="radio"
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: May we can use Bootstrap markup to inherit of common radio buttons style

File: dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header-controller.js (hunk 4 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/lookup/grid/header/lookup-datagrid-header-controller.js]: :+1:

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html (hunk 7 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: :+1:

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js (hunk 6 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu-controller.js]: Fixed, thanks

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html (hunk 8 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: Fixed, thanks

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html (hunk 7 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: `translate-once` can only be used for static text : https://github.com/ajwhite/angular-translate-once/blob/develop/README.md

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html [hunk 9]
```diff
@@ -19,22 +19,33 @@
...
-            />
-            {{ ::semanticDomain.label }} {{ semanticDomain.frequency }} %
+            <div class="radio">
+              <label>
+                  <input  type="radio"
+                          value="{{ ::semanticDomain.id }}"
+                          ng-model="$ctrl.currentDomain"
+                          ng-checked="$ctrl.currentDomain === semanticDomain.label"
+                  />
+                  {{ ::semanticDomain.label }} {{ semanticDomain.frequency }} %
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: you need span here for some browsers

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html [hunk 10]
```diff
@@ -19,15 +19,33 @@
...
+
+            <div class="radio">
+              <label>
+                  <input  type="radio"
+                          value="{{ ::type.id}}"
+                          ng-model="$ctrl.currentDomain"
+                          ng-checked="$ctrl.shouldBeChecked(type)"
+                  />
+                  <span class="set-type-label" translate-once="COLUMN_TYPE_SET" />
+                  <span class="type">{{ ::type.labelKey | translate }}</span>
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: You can use translate-once here too and wrap these to span into one to constitute a real label

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/448/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/448/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/450/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/450/)

File: dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html (hunk 10 above)
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/type/type-transformation-menu.html]: I can't, `translate-once`should be use for static text only.

**Comment by frassinier** [comment]
I cannot change semantic type any more :(
peek 30-06-2017 17-04

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/462/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/462/)

**Comment by ncomont** [comment]
@frassinier I cannot reproduce this issue. Can you ensure that you tested on an up-to-date branch, please ? This bug was fixed 5 days ago by this commit : https://github.com/Talend/data-prep/pull/690/commits/e6475ae429db037957ba444984162de811da4977 (the semantic type is changed, but the label is not returned).

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/471/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/471/)

**Pull request merged**